frontend/taipy-gui/src/components/Taipy/Metric.tsx

```diff
@@ -66,6 +67,22 @@ const Metric = (props: MetricProps) => {
     const hover = useDynamicProperty(props.hoverText, props.defaultHoverText, undefined);
     const theme = useTheme();
 
+    const colorMap = useMemo(() => {
+        try {
+            const obj = props.colorMap ? JSON.parse(props.colorMap) : null;
+            if (obj && typeof obj === 'object') {
+                const keys = Object.keys(obj);
+                return keys.sort((a, b) => Number(a) - Number(b)).map((key, index) => {
+                    const nextKey = keys[index + 1] !== undefined ? Number(keys[index + 1]) : props.max || 100;
+                    return {range: [Number(key), nextKey], color: obj[key]};
+                }).filter(item => item !== null && item.color !== null)
```

Comment:
How could item be null ?

Reply:
At some point, I just want to make sure that the returned list does not contain any null items, but I guess it is not necessary here.